pymc3/model.py

```diff
@@ -913,16 +912,26 @@ def independent_vars(self):
 
     @property
     def test_point(self):
+        """Deprecated alias for `Model.initial_point`."""
         warnings.warn(
             "`Model.test_point` has been deprecated. Use `Model.initial_point` instead.",
             DeprecationWarning,
         )
         return self.initial_point
 
     @property
-    def initial_point(self):
+    def initial_point(self) -> Dict[str, np.ndarray]:
+        """Maps names of variables to initial values."""
         return Point(list(self.initial_values.items()), model=self)
 
+    @property
+    def initial_values(self) -> Dict[TensorVariable, np.ndarray]:
+        """Maps transformed variables to initial values.
+
+        ⚠ The keys are NOT the objects returned by, `pm.Normal(...)`.
+        """
+        return self._initial_values
+
```

Comment:
Why is this being purposefully exposed?  Do we expect that users will need access to this particular map?  

Remember, when you start exposing things like this, it's more likely that code outside of the PyMC3 internals will start to use/rely on it, and the internal initial value handling doesn't look mature enough to risk that.

Reply:
The dictionary was exposed already, I just made it a property so I could attach type hints and docstring.

After working with `Model.initial_values` and `Model.initial_point` for a few hours now I must agree that we shouldn't expose them. (We can make it a "private" property too.)
I was very confused by the keys in these dictionaries not having anything to do with the RVs returned at the user level API.
Accordingly the values in there are in terms of transformed initvals too..

Reply:
> The dictionary was exposed already, I just made it a property so I could attach type hints and docstring.

Making it a property with type hints and a docstring is the type of "exposing" that I'm talking about.  If we build more stuff around these things, then they tend to become harder to refactor and more likely to be used elsewhere.

Reply:
We can make them private, but that's not in the scope of this PR.
We should then also discuss if both of `Model.init_values`, `Model.init_point` should be private.

Again, I'm all in favor of making them private, but to my understanding that's a matter of naming and not of docstrings and type hints, which we should have also on internal things for own good.